core/jsMain/src/Instant.kt

```diff
@@ -105,40 +105,14 @@ public actual fun Instant.plus(period: DateTimePeriod, timeZone: TimeZone): Inst
                 .run { if (days != 0) plusDays(days) as ZonedDateTime else this }
                 .run { if (hours != 0) plusHours(hours) else this }
                 .run { if (minutes != 0) plusMinutes(minutes) else this }
-                .run { plusSecondsFix(seconds) }
-                .run { plusNanosFix(nanoseconds) }
+                .run { if (seconds != 0L) plusSeconds(seconds.toDouble()) else this }
+                .run { if (nanoseconds != 0L) plusNanos(nanoseconds.toDouble()) else this }
```

Comment:
Why is `toDouble` needed here? Tests at least pass successfully even without it.